src/module.cpp

```diff
@@ -1075,6 +1075,16 @@ void Module::AddFunctionDeclaration(const std::string &name, const FunctionType
     auto [name_pref, name_suf] = functionType->GetFunctionMangledName(false);
     std::string functionName = name_pref + name + name_suf;
 
+    for (int i = 0; i < functionType->GetNumParameters(); i++) {
+        const Type *t = functionType->GetParameterType(i);
+        if (!t->IsCompleteType()) {
+            const SourcePos paramPos = functionType->GetParameterSourcePos(i);
+            const std::string paramName = functionType->GetParameterName(i);
+            Error(paramPos, "parameter %d ('%s') has incomplete type '%s'", i, paramName.c_str(),
```

Comment:
I think just reporting of parameter name with its source pos is sufficient, no index is needed. It looks too details as some debug info.